File: Teste.Shared/Models/OrderModel.cs
using System.Text.Json.Serialization;
using Teste.Shared.Enums;

namespace Teste.Shared.Models;

public sealed class OrderModel
{
    [JsonPropertyName("identificador")]
    public Guid OrderId { get; set; }
    [JsonPropertyName("dataVenda")]
    public DateTime OrderDate { get; set; }
    [JsonPropertyName("cpf")]
    public CustomerModel Customer { get; set; } = default!;
    [JsonPropertyName("categoria")]
    public IEnumerable<OrderItemModel> OrderItems { get; set; } = default!;
}

File: Teste.Web/src/models/order-model.ts
import { OrderStatus } from "@/enums/order-status";
import { CustomerModel } from "@/models/customer-model";
import { OrderItemModel } from "@/models/order-item-model";

export interface OrderModel {
  identificador: string;
  dataVenda: string;
  cliente: CustomerModel;
  itens: OrderItemModel[];
  subtotal: number;
  desconto: number;
  valorTotal: number;
  status: OrderStatus;
}
